tests/test_tracking_grid/test_tracking_grid.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python
+
+import os
+import sys
+import math
+sys.path.insert(0, os.pardir)
+sys.path.insert(0, os.path.join(os.pardir, 'openmoc'))
+from testing_harness import TestHarness
+from input_set import GridInput
+
+import openmoc
+
+class TrackingGridTestHarness(TestHarness):
+    """Tests tracking over a grid geometry."""
+
+    def __init__(self):
+        super(TrackingGridTestHarness, self).__init__()
+        self.input_set = GridInput()
+        self._result = ''
+
+    def _setup(self):
+        """Initialize the materials and geometry in the InputSet."""
+        super(TrackingGridTestHarness, self)._create_geometry()
+
+    def _segment_track(self, track, geometry):
+        """Segments a given track over a given geometry and records the
+           resulting segment information to a string"""
+
+        # Segmentize a track in a geometry, recording the segments in a string
+        geometry.segmentize(track)
+        num_segments = track.getNumSegments()
+        info = ' ' + str(num_segments) + '\n'
+        for i in range(num_segments):
+            info += str(i) + ': '
+            segment = track.getSegment(i)
+            info += str(round(segment._length, 8)) + ', '
+            info += str(segment._region_id) + ', '
+            info += str(segment._cmfd_surface_fwd) + ', '
+            info += str(segment._cmfd_surface_bwd) + ', '
+            info += str(segment._material.getName()) + ', '
```

Comment:
Material ID is sufficient for checking accuracy, material name helps in debugging. The name is not sufficient because multiple materials could potentially have the same name.